File: exchange_rate_app_react/src/model/ForecastComparisonApiResponse.ts
import {CurrencyForecast} from "./CurrencyForecast";

export class ForecastComparisonApiResponse {
    success: boolean;
    forecastComparisons: ForecastComparisonApiResponse.ForecastItem[];

    constructor(success: boolean, forecastComparisons: ForecastComparisonApiResponse.ForecastItem[]) {
        this.success = success;
        this.forecastComparisons = forecastComparisons;
    }
}

export namespace ForecastComparisonApiResponse {
    export class ForecastItem {
        conversionRate: number;
        forecastData: CurrencyForecast;

        constructor(conversionRate: number, forecastData: CurrencyForecast) {
            this.conversionRate = conversionRate;
            this.forecastData = forecastData;
        }
    }
}

File: exchange_rate_app_swift/exchange_rate_app_swift/model/ForecastComparisonApiResponse.swift
//
//  ForecastComparisonApiResponse.swift
//  exchange_rate_app_swift
//
//  Created by Anthony Gibah on 9/4/24.
//

import Foundation

struct ForecastComparisonApiResponse: Codable {
    let success: Bool
    let forecastComparisons: [ForecastItem]
}


struct ForecastItem: Codable {
    let conversionRate: Double
    let forecastData: CurrencyForecast
}

func decodeForecastComparisonData(from jsonData: Data) -> ForecastComparisonApiResponse? {
    let decoder = JSONDecoder()
    decoder.dateDecodingStrategy = .iso8601 // Adjust if needed
    do {
        let forecastData = try decoder.decode(ForecastComparisonApiResponse.self, from: jsonData)
        return forecastData
    } catch {
        print("Failed to decode JSON: \(error)")
        return nil
    }
}
